modules/dashboards/application/controllers/DashletsController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Icinga\Module\Dashboards\Controllers;
+
+use Icinga\Module\Dashboards\Form\NewDashletsForm;
+use Icinga\Module\Dashboards\Model\Database;
+use Icinga\Module\Dashboards\Web\Controller;
+use Icinga\Module\Dashboards\Web\Widget\DashletWidget;
+use ipl\Sql\Select;
+
+class DashletsController extends Controller
+{
+    use Database;
+
+    private $data;
+
+    public function indexAction()
+    {
+        $selectDashboard = (new Select())
+            ->columns('*')
+            ->from('dashboard');
+
+        $this->data = $this->getDb()->select($selectDashboard);
+
+        foreach ($this->data as $dashboard) {
+            $this->setTitle($this->translate($dashboard['name']));
+        }
+
+        $divContent = $this->content;
+        $divContent->setAttributes(['class' => 'dashboard content']);
+
+        $selectDashlet = (new Select())
+            ->columns('dashlet.name, dashlet.url')
+            ->from('dashlet')
+            ->join('dashboard d', 'dashlet.dashboard_id= d.id');
+
+        $this->data = $this->getDb()->select($selectDashlet);
+        $dashlets = new DashletWidget($this->data);
+
+        $this->addContent($dashlets);
+    }
+
+    public function newAction()
+    {
+        $this->setTitle('New Dashlet');
+
+        $dashletsForm = new NewDashletsForm();
+        $this->addContent($dashletsForm);
+
+        if ($this->getRequest()->isPost()) {
```

Comment:
You don't have to manually implement the form handling process which includes

1. Comparing whether the request method equals the form's method
2. Detecting whether the submit button has been pressed
3. Validating the form

You barely do step one 😆 

But that is what `Form::onSuccess()` is for. Please move the database code there.
`Form ::onSuccess()` is called if all of the above is valid.

Note that the process for handling forms is triggered via `Form::handleRequest()`. You have to call that in the controller.

In controllers, also, you do the redirect, i.e.

```php
$dashletForm->on($dashletForm::ON_SUCCESS, function () {
    $this->redirectNow('dashboards/dashlets');
});
```

The `ipl/form` implementation supports events and we're listening for the `Form::ON_SUCCESS` event here.
That event is fired after `onSuccess` is called. Note that the same validation steps from above apply to this event.